components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -555,7 +559,21 @@ protected void doAuthentication(HttpServletRequest request, HttpServletResponse
             authenticatedIdPData.addAuthenticator(authenticatorConfig);
             //add authenticated idp data to the session wise map
             context.getCurrentAuthenticatedIdPs().put(idpName, authenticatedIdPData);
-            context.addAuthenticationStepHistory(new AuthHistory(authenticator.getName(), idpName));
+
+            // Add SAML federated idp session index into the authentication step history.
+            String idpSessionIndex = null;
+            String parameterName = FEDERATED_IDP_SESSION_ID + idpName;
+            AuthHistory authHistory = new AuthHistory(authenticator.getName(), idpName);
+
+            if (context.getParameters().containsKey(parameterName)) {
```

Comment:
Can't `context.getParameters()` e null? if so do the null check

Reply:
Fixed with PR https://github.com/wso2/carbon-identity-framework/pull/3738